paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/validation/PaytrailPaymentReturnValidator.java

```diff
@@ -54,11 +72,21 @@ private boolean isSuccesfull(String status) {
         return PaymentStatus.OK.getStatus().equals(status);
     }
 
-    private String buildSignatureFromReturnData(Map<String,String> checkoutParams) {
+    private String buildSignatureFromReturnData(Map<String,String> checkoutParams, String merchantId, boolean isShopInShopPayment) {
```

Comment:
Signature calculation will require the merchant specific secret key if we use normal merchant flow.